servers/graphql-kotlin-ktor-server/src/main/kotlin/com/expediagroup/graphql/server/ktor/GraphQLConfiguration.kt

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2023 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.server.ktor
+
+import com.expediagroup.graphql.dataloader.KotlinDataLoaderRegistryFactory
+import com.expediagroup.graphql.generator.TopLevelNames
+import com.expediagroup.graphql.generator.execution.KotlinDataFetcherFactoryProvider
+import com.expediagroup.graphql.generator.execution.SimpleKotlinDataFetcherFactoryProvider
+import com.expediagroup.graphql.generator.hooks.NoopSchemaGeneratorHooks
+import com.expediagroup.graphql.generator.hooks.SchemaGeneratorHooks
+import com.expediagroup.graphql.generator.scalars.IDValueUnboxer
+import com.expediagroup.graphql.server.Schema
+import com.expediagroup.graphql.server.execution.GraphQLRequestHandler
+import com.expediagroup.graphql.server.operations.Mutation
+import com.expediagroup.graphql.server.operations.Query
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import graphql.execution.DataFetcherExceptionHandler
+import graphql.execution.ExecutionIdProvider
+import graphql.execution.SimpleDataFetcherExceptionHandler
+import graphql.execution.instrumentation.Instrumentation
+import graphql.execution.preparsed.PreparsedDocumentProvider
+import io.ktor.server.config.ApplicationConfig
+import io.ktor.server.config.tryGetString
+import io.ktor.server.config.tryGetStringList
+
+/**
+ * Configuration properties that define supported GraphQL configuration options.
+ */
+class GraphQLConfiguration(config: ApplicationConfig)
```

Comment:
I started with the same configuration properties as used by Spring. This config contains some properties that can be set through configuration files (i.e. same as spring) and others that can only be set programmatically.

I'm not a big fan of how this is all structured at the moment so most likely structure of this file will change.

In general there are 3 areas of configuration:
- schema - everything related to schema generation (e.g. which queries to include)
- engine - everything related to GraphQL Java engine configuration
- server - Ktor server custom config (e.g. context factory)

I'm thinking I will move the existing properties (which were matching spring config) under those three sections, e.g. when configuring schema I think it would make sense to specify target packages in the same place

Reply:
One drawback of reorganizing the properties would be the change in property file mapping - currently properties match the path from the root `graphql` prefix, e.g. `graphql.packages` is a top level `packages` property. If we move those around then we either have to use different properties than in Spring (not ideal but I guess it should be fine?) or have a mismatch between paths and variables, e.g. 

* `graphql.packages` would become `graphql.schema.packages`
* `graphql.introspection.enabled` would become `graphql.engine.introspection.enabled`